tasks/build-binary-new/builder.rb

```diff
@@ -18,12 +18,29 @@ def run(*args)
 
 module DependencyBuild
   class << self
+
+    def self.build_python_for_pip(python_version)
+     # The python_version argument is the version of python that we are building for pip. It requires major, minor and
+     # patch. Eg: 3.7.4
+
+     Runner.run('apt', 'update')
+     Runner.run('apt', 'install', '-y', 'build-essential', 'zlib1g-dev', 'libncurses5-dev', 'libgdbm-dev', 'libnss3-dev', 'libssl-dev', 'libsqlite3-dev', 'libreadline-dev', 'libffi-dev', 'curl' ,'libbz2-dev')
+     Runner.run('mkdir', '-p', '/tmp/python/')
+     Runner.run('curl', '-L', 'https://www.python.org/ftp/python/' + python_version + '/Python-' + python_version + '.tgz', '-o', '/tmp/python/Python-' + python_version + '.tgz')
+     Runner.run('tar', '-xzf', '/tmp/python/Python.tgz', '-C', '/tmp/python/')
+     Runner.run('cd', '/tmp/python/Python-' + python_version + ' && ./configure && make && make install')
+    end
+
     def bundle_pip_dependencies(source_input)
       # final resting place for pip source and dependencies
       file_path = "/tmp/pip-#{source_input.version}.tgz"
       ENV['LC_CTYPE'] = 'en_US.UTF-8'
-      Runner.run('apt', 'update')
-      Runner.run('apt-get', 'install', '-y', 'python3-pip')
+
+      # For the latest version of pip, it requires python version >= 3.7 (ref: https://github.com/pypa/pip/pull/10641),
+      # so we need to build python >= 3.7 first.
+
+      build_python_for_pip('3.7.11')
```

Comment:
I'm concerned that hardcoding version to a patch would mean that we have a high chance of missing out on critical security fixed.
Any reason we can't replace the function with something like the following example?
```bash
apt update
# or any 3.x version you like
apt-get install -y python3.8
# python3.8 --version
# Python 3.8.0
```

Reply:
When using `apt install python3.7` pip is not installed, so it must be configured manually using [this file](https://bootstrap.pypa.io/get-pip.py). This requires another package called `python3.7-distutils` to be installed.

Do you think it is clearer to install Python/Pip using this method?

Reply:
@arjun024 PR Updated!